components/browser/menu/src/main/res/values/style.xml

```diff
@@ -3,6 +3,11 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <resources>
+
+    <style name="Mozac.Browser.Menu" parent="">
+        <item name="cardBackgroundColor">?attr/menuBackgroundColor</item>
```

Comment:
This is could cause a silent run time crash, If the app doesn't override the `Mozac.Browser.Menu` style and define the value of `?attr/menuBackgroundColor`.

To see the issue, run the `samples-browsers`app  and tap the 3 dots menu:

An alternative could be instead of using the attribute `menuBackgroundColor`, we could have a default color and let the app override it.

Something like:

**menu/colors.xml**
```xml
<resources>
    <color name="mozac_browser_menu_background">#E91E63</color>
</resources>
```
**menu/style.xml**
```xml
 <style name="Mozac.Browser.Menu" parent="">
      <item name="cardBackgroundColor">@color/mozac_browser_menu_background</item>
</style>
```

In the app
**color.xml**
```xml
<resources>
    <color name="mozac_browser_menu_background" tools:ignore="UnusedResources">YOU_COLOR</color>
</resources>
```
If you go with this approach please update the [docs](https://github.com/mozilla-mobile/android-components/blob/master/components/browser/menu/README.md) :)

```
2019-02-21 11:33:58.082 4476-4476/org.mozilla.samples.browser E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.samples.browser, PID: 4476
    android.view.InflateException: Binary XML file line #5: Binary XML file line #5: Error inflating class android.support.v7.widget.CardView
    Caused by: android.view.InflateException: Binary XML file line #5: Error inflating class android.support.v7.widget.CardView
    Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at android.view.LayoutInflater.createView(LayoutInflater.java:647)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:790)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
        at mozilla.components.browser.menu.BrowserMenu.show(BrowserMenu.kt:31)
        at mozilla.components.browser.toolbar.display.DisplayToolbar$$special$$inlined$apply$lambda$2.onClick(DisplayToolbar.kt:121)
        at android.view.View.performClick(View.java:6597)
        at android.view.View.performClickInternal(View.java:6574)
        at android.view.View.access$3100(View.java:778)
        at android.view.View$PerformClick.run(View.java:25885)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.UnsupportedOperationException: Failed to resolve attribute at index 2: TypedValue{t=0x2/d=0x7f03017b a=3}
        at android.content.res.TypedArray.getColorStateList(TypedArray.java:546)
        at android.support.v7.widget.CardView.<init>(CardView.java:127)
        at android.support.v7.widget.CardView.<init>(CardView.java:117)
        at java.lang.reflect.Constructor.newInstance0(Native Method) 
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343) 
        at android.view.LayoutInflater.createView(LayoutInflater.java:647) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:790) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:492) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:374) 
        at mozilla.components.browser.menu.BrowserMenu.show(BrowserMenu.kt:31) 
        at mozilla.components.browser.toolbar.display.DisplayToolbar$$special$$inlined$apply$lambda$2.onClick(DisplayToolbar.kt:121) 
        at android.view.View.performClick(View.java:6597) 
        at android.view.View.performClickInternal(View.java:6574) 
        at android.view.View.access$3100(View.java:778) 
        at android.view.View$PerformClick.run(View.java:25885) 
        at android.os.Handler.handleCallback(Handler.java:873) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6718) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
    

    --------- beginning of system
```

Reply:
Good catch on this. I'm still a bit new to applying themes through XML and struggle with understanding the linking--this was really helpful!

Reply:
:)

Reply:
Does the color approach really work since Fenix uses different themes and therefore needs to support multiple colors.

Reply:
@pocmo I think the problem is even if you *don't* override the style you have a crash. So you're required to set a style for it. It still seems like a nicer solution than overriding a color, but it seems confusing that a user would have to set it.

Reply:
@pocmo  if you `mozac_browser_menu_background` you will only need to change the color in the app, but if you override the style you will have to define all the values.

Reply:
I suggested the approach of override the color, because if you override the style you will have to define all the values, even the ones that you are not interested in changing.

For instance, we  have:

**In AC**
```XML
 <style name="Mozac.Browser.Menu" parent="">
      <item name="cardBackgroundColor">#A87687</item>
        <item name="android:paddingEnd">10dp</item>
</style>
```

**In the app:**
```XML
 <style name="Mozac.Browser.Menu" parent="">
      <item name="cardBackgroundColor">NEW_COLOR</item>
</style>
```

The `paddingEnd` won't be applied just the `cardBackgroundColor`.

If want to do both you have to copy the complete style from **AC**

 ```xml
<style name="Mozac.Browser.Menu" parent="">
      <item name="cardBackgroundColor">NEW_COLOR</item>
      <item name="android:paddingEnd">10dp</item>
</style>
```